CDCgov/prime-reportstream
# Router Refactor: Update IA page framework to adopt v6

> **NOTE**: This is part of a larger refactor, and is merging safely into a `shadow` branch!

This PR works through IA framework files to refactor routing patterns for v6.

Test Steps:
1. Pull this branch and run `yarn` to install new dependencies
2. Load the site
3. Ensure the public site has no broken page navigation (content links do not apply, navigation only)

## Changes
- Removes `GeneratedRouter` as all generated routers must be in-line
- Refactors `IAMetaAndRouter` into separate components for composition
- Introduces `IATemplate`, the main interface for templating your IA directory (see example in `Resources.tsx`)
- Minor refactors to address relative paths, names, and more in IA framework

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
